src/classes/DateLiteralConstant.cls

```diff
@@ -0,0 +1,225 @@
+public class DateLiteralConstant {
+    private String LAST_N = 'LAST_N_{0}: {1}';
+    private String NEXT_N = 'NEXT_N_{0}: {1}';
+
+    private final String DAYS            = 'DAYS';
+    private final String WEEKS           = 'WEEKS';
+    private final String MONTHS          = 'MONTHS';
+    private final String QUARTERS        = 'QUARTERS';
+    private final String FISCAL_QUARTERS = 'FISCAL_QUARTERS';
+    private final String YEARS           = 'YEARS';
+    private final String FISCAL_YEARS    = 'FISCAL_YEARS';
+
+    public String value {get;set;}
+
+    public DateLiteralConstant(){}
+
+    private DateLiteralConstant(String value) {
```

Comment:
I've thought of 1 option that I think I like slightly better? But let me know what you think.

To try to simplify my examples, I'll just use the TODAY literal as an example.

```
public DateLiteralConstant TODAY {
    get {
        // Call a new setValue() method
        return this.setValue('TODAY'));
    }
}

// New method that sets the value & return the current instance of the class, instead of creating a new instance
private DateLiteralConstant setValue(String value) {
    this.value = value;
    return this;
}
```